### PR TITLE
Add pre-commit instructions to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by hiaselhans**
Signed-off-by: hiaselhans <simon.klemenc@gmail.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -93,6 +93,23 @@ for more information.
   commit message title. To a reasonable extent, additional tags can be used
   to signify the component changed, e.g.: `[PI]`, `[CUDA]`, `[Doc]`.
 
+### Pre-commit checks
+
+- Before submitting your patch you can verify coding-style conformance with the clang-format tool:
+  ```
+  git diff -U0 sycl~0 --no-color -- | ./clang/tools/clang-format/clang-format-diff.py -p1
+  ```
+  make sure clang-format is installed on your system
```
Comment (AGindinson) [inline CONTRIBUTING.md:102]: Could this be merged with lines 81-84?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -93,6 +93,23 @@ for more information.
...
+
+- You can also run the sycl test-suite before sending your PR:
+
+    - complete test suite: `python buildbot/check.py`
+    - sycl test suite `python buildbot/check.py -t test-sycl`
+    - run only "mytest" test `python buildbot/check.py -t test-sycl-mytest`
+  
+  if necessary, use `-o $LLVM_BUILD_DIR` to specify the llvm build directory
```
Comment (AGindinson) [inline CONTRIBUTING.md:110]: Could this be placed after line 85?

File: CONTRIBUTING.md (hunk 1 above)
Comment (hiaselhans) [reply CONTRIBUTING.md:102]: you are right! sorry.
i will integrate it into what's there already. just thought we could provide a lookup for pre-commit commands.

Comment (AGindinson) [reply CONTRIBUTING.md:102]: Sure, the commands are really useful.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -77,12 +77,21 @@ for more information.
...
-  https://clang.llvm.org/extra/clang-tidy/) tools can be integrated into your
-  workflow to ensure formatting and stylistic compliance of your changes.
+- Prepare your patch
+    - follow [LLVM coding standards](https://llvm.org/docs/CodingStandards.html)
+    - [clang-format](https://clang.llvm.org/docs/ClangFormat.html) and 
+      [clang-tidy](https://clang.llvm.org/extra/clang-tidy/) tools can be integrated into your
+      workflow to ensure formatting and stylistic compliance of your changes.
+    - use
+      ```
+      git diff -U0 sycl~0 --no-color -- | ./clang/tools/clang-format/clang-format-diff.py -p1
```
[the remark this answers is not in the document]
Comment (hiaselhans) [reply CONTRIBUTING.md]: done, thx @bader 
most of the time origin/sycl is also just the remote branch of the fork, but everyone should be good enough to realize this. It's only meant as a quick hint after all...

**Comment by bader** [comment]
Thanks!

**Pull request merged**